server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -117,21 +121,24 @@ private GetMailboxesResponse getMailboxesResponse(GetMailboxesRequest mailboxesR
 
     private Stream<Mailbox> retrieveMailboxes(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
         return mailboxIds
-            .map(ids -> retrieveSpecificMailboxes(mailboxSession, ids))
+            .map(Throwing.function((ImmutableList<MailboxId> ids) -> retrieveSpecificMailboxes(mailboxSession, ids)).sneakyThrow())
             .orElseGet(Throwing.supplier(() -> retrieveAllMailboxes(mailboxSession)).sneakyThrow());
     }
 
-    private Stream<Mailbox> retrieveSpecificMailboxes(MailboxSession mailboxSession, ImmutableList<MailboxId> mailboxIds) {
+    private Stream<Mailbox> retrieveSpecificMailboxes(MailboxSession mailboxSession, ImmutableList<MailboxId> mailboxIds) throws MailboxException {
+        Quotas quotas = quotaLoader.getQuotas(mailboxSession.getUser());
         return mailboxIds
             .stream()
             .map(mailboxId -> mailboxFactory.builder()
                 .id(mailboxId)
+                .withUserQuota(quotas)
```

Comment:
do you loose per mailbox quota, or did I miss somethnig?

Reply:
In JMAP, yes

Reply:
which quota ratio API? As far as I understand, Inbox choose to only refer to INBOX's quota to know about the ratio. But until their there is a different quota on each mailbox.

Reply:
and what about shared mailboxes?